### PR TITLE
Separate scan lines filter

### PR DESCRIPTION
**Pull request by gui2dev**
This is a basic implementation for filter that separates scan lines, using the ``EdgeOfFlightLine`` dimension.

### CODE REVIEW

File: filters/SeparateScanLineFilter.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+{
+    PointLayoutPtr layout(table.layout());
+    if (!layout->hasDim(Dimension::Id::EdgeOfFlightLine))
+        throwError("Layout does not contains EdgeOfFlightLine dimension.");
+}
+
+PointViewSet SeparateScanLineFilter::run(PointViewPtr inView)
+{
+    PointViewSet result;
+    std::vector<PointViewPtr> views;
```
Comment (abellgithub) [inline filters/SeparateScanLineFilter.cpp]: Think this is unused.

File: filters/SeparateScanLineFilter.hpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+class PDAL_DLL SeparateScanLineFilter : public Filter
+{
+public:
+    SeparateScanLineFilter();
+
+    std::string getName() const;
+
+private:
+    std::map<uint64_t, PointViewPtr> m_viewMap;
```
Comment (abellgithub) [inline filters/SeparateScanLineFilter.hpp]: Unused

File: filters/SeparateScanLineFilter.hpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    uint64_t m_groupBy;
```
Comment (abellgithub) [inline filters/SeparateScanLineFilter.hpp]: The type of this variable should be the same as the type of `i` in the processing loop.

File: filters/SeparateScanLineFilter.hpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    virtual void addArgs(ProgramArgs& args);
+    virtual void prepared(PointTableRef table);
+    virtual PointViewSet run(PointViewPtr view);
+
+    SeparateScanLineFilter& operator=(const SeparateScanLineFilter&); // not implemented
```
Comment (abellgithub) [inline filters/SeparateScanLineFilter.hpp]: If you could move these to be tagged as `= deleted` that would be great.

Comment (gui2dev) [reply filters/SeparateScanLineFilter.hpp]: @abellgithub, not sure what you mean

Comment (abellgithub) [reply filters/SeparateScanLineFilter.hpp]: NP  I'll take care of it.  Thanks again.

**Pull request merged**